File: questions/1658. Minimum Operations to Reduce X to Zero/min.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int res = INT_MIN, target = -x, sum = 0;
        for(int i : nums){
            target += i;
        }
        if(target == 0) return nums.size();
        unordered_map<int, int> m;
        m[0] = -1;
        for(int i = 0; i < nums.size(); i++){
            sum += nums[i];
            if(m.count(sum - target)){
                res = max(res, i - m[sum - target]);
            }
            m[sum] = i;
        }
        return res == INT_MIN?-1:nums.size() - res;
    }
};

File: questions/1658. Minimum Operations to Reduce X to Zero/min.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        s, res, target = 0, -float('inf'), -x       
        for i in nums:
            target += i
        if not target:
            return len(nums)
        dic = defaultdict(int)
        dic[0] = -1
        for i in range(len(nums)):
            s += nums[i]
            if s - target in dic:
                res = max(res, i - dic[s - target])
            dic[s] = i
        return len(nums) - res if res != -float('inf') else -1
        
